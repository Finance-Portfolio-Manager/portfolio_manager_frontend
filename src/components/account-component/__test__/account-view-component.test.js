import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import AccountView from './../account-view-component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let portObj1 = {id: 1, name:"Portfolio 1", value:1000};
let portObj2 = {id: 2, name:"Portfolio 2", value:145641};
let portObj3 = {id: 3, name:"Portfolio 3", value:1840};
let portfolioArray = [portObj1, portObj2, portObj3];

const wrapper = shallow(<AccountView portfolioArray={ portfolioArray }></AccountView>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<AccountView portfolioArray={ portfolioArray }></AccountView>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("username header renders", ()=>{
    expect(wrapper.find("h2").text()).toContain("Username");
})