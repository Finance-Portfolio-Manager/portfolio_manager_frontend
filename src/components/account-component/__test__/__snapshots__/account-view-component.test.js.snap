// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#B9B9BA",
    }
  }
>
  <h2
    id="account-header"
  >
    <span
      id="header-span"
    >
      Username
    </span>
  </h2>
  <div
    className="row"
  >
    <div
      className="col-md-2 col-6"
    >
      <img
        className="rounded"
        src="exampleimg.svg"
      />
    </div>
    <div
      className="col"
    >
      <div
        className="center card"
        style={
          Object {
            "border": "2px solid #F26925",
            "width": "50rem",
          }
        }
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Portfolio 1
          </div>
          <p
            className="card-text"
          >
            Current value: 
            1000
             %change placeholder
          </p>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Details
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-2"
    />
  </div>
   
  <br />
  <div
    className="row"
  >
    <div
      className="col-md-2 col-6"
    >
      <img
        className="rounded"
        src="exampleimg.svg"
      />
    </div>
    <div
      className="col"
    >
      <div
        className="center card"
        style={
          Object {
            "border": "2px solid #F26925",
            "width": "50rem",
          }
        }
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Portfolio 2
          </div>
          <p
            className="card-text"
          >
            Current value: 
            145641
             %change placeholder
          </p>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Details
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-2"
    />
  </div>
   
  <br />
  <div
    className="row"
  >
    <div
      className="col-md-2 col-6"
    >
      <img
        className="rounded"
        src="exampleimg.svg"
      />
    </div>
    <div
      className="col"
    >
      <div
        className="center card"
        style={
          Object {
            "border": "2px solid #F26925",
            "width": "50rem",
          }
        }
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Portfolio 3
          </div>
          <p
            className="card-text"
          >
            Current value: 
            1840
             %change placeholder
          </p>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Details
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-2"
    />
  </div>
   
  <br />
</div>
`;
